module/VuFind/src/VuFind/I18n/Translator/Loader/ExtendedIni.php

```
@@ -181,30 +227,165 @@ protected function checkAndMarkLoadedFile($filename)
         // unexpected, fatal error; a missing domain-specific file is more likely
         // due to the possibility of incomplete translations.
-        return $this->loadLanguageFile($filename, empty($domain));
+        return $this->loadLanguageFile(
+            $filename,
+            empty($domain),
+            $processAliases ? (empty($domain) ? 'default' : $domain) : null
+        );
+    }
+
+    /**
+     * Resolve a single alias (or return null if it cannot be resolved)
+     *
+     * @param array  $alias         The [domain, key] or [key] alias array
+     * @param string $defaultDomain The domain to use if $alias does not specify one
+     * @param string $locale        The locale currently being loaded
+     * @param array  $breadcrumbs   Previously-resolved aliases (to prevent infinite loops)
+     *
+     * @return ?string
+     */
+    protected function resolveAlias(
+        array $alias,
+        string $defaultDomain,
+        string $locale,
+        array $breadcrumbs = []
+    ): ?string {
+        // If the current alias target does not include a TextDomain part, assume it refers
+        // to the current active TextDomain:
+        if (count($alias) < 2) {
+            array_unshift($alias, $defaultDomain);
+        }
+        [$domain, $key] = $alias;
+
+        // If the alias references another TextDomain, we need to load that now.
+        if (!isset($this->aliasDomains[$domain])) {
+            $this->aliasDomains[$domain] = $this->loadLanguageLocale($locale, $domain, true);
+        }
+        if ($this->aliasDomains[$domain]->offsetExists($key)) {
+            return $this->aliasDomains[$domain]->offsetGet($key);
+        } elseif (isset($this->aliases[$domain][$key])) {
+            // Circular alias infinite loop prevention:
+            $breadcrumbKey = "$domain::$key";
+            if (in_array($breadcrumbKey, $breadcrumbs)) {
+                return null;
```

Would it be better to throw an exception here? Otherwise finding the error might be hard.

---

Yes, very good point. I have changed the behavior in [721c4a9](https://github.com/vufind-org/vufind/pull/3249/commits/721c4a985dbcab0bbbbd3abce05fafdd17c76ff9).